### ISSUE TITLE
[feat] inject conventional commit message type before issue number

### ISSUE THREAD

**Issue by WuglyakBolgoink** [nemwiz/jira-commit-message-intellij-plugin#64]
**Describe the solution you'd like**

if we use in branch naming the conventional commits type, it will be great to add this automatically

```
Type
Only certain types are permitted, with the most common being:

fix: a commit that fixes a bug in Uno.
feat: a commit that adds new functionality to Uno.
docs: a commit that adds or improves Uno's documentation.
test: a commit that adds unit tests.
perf: a commit that improves performance, without functional changes.
chore: a catch-all type for any other commits. For instance, if you're implementing a single feature and it makes sense to divide the work into multiple commits, you should mark one commit as feat and the rest as chore.
```

See: https://platform.uno/docs/articles/uno-development/git-conventional-commits.html#type

Example: 

branch name: `feat-MYKEY-1234-someadditionalödescriptiontobranch`
result with `infix`: `feat(MYKEY-1234):`

Additionally please add in JB Plugin Interface new setting: **"Automatically detect Conventional Commits: [ ]"** analog to **"JIRA project key"**

**Comment by nemwiz** [comment]
Hey @WuglyakBolgoink,

Sounds like a great idea. Conventional commits are not tied to Jira (which is the main purposes of this plugin), but I see the benefit of having this option in the plugin itself. Adding another optional checkbox would probably be enough, but let me think a little bit on this and I'll write up a solution.

**Comment by nemwiz** [comment]
@WuglyakBolgoink 
I've implemented this feature. Could you test it out? Let me know if you encounter any issues.

You can install this version by going to Plugins -> Click the cog icon -> Install plugin from disk
[jira-commit-message-3.3.0.zip](https://github.com/nemwiz/jira-commit-message-intellij-plugin/files/11358035/jira-commit-message-3.3.0.zip)

Thanks

**Comment by nemwiz** [comment]
Thanks for suggesting this feature. It has been released in [v3.3.0](https://github.com/nemwiz/jira-commit-message-intellij-plugin/releases/tag/v3.3.0)
Please leave [a review](https://plugins.jetbrains.com/plugin/14310-jira-id-commit-message) if you find this plugin useful. Thanks
